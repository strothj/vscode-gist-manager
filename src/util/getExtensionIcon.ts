import fs from "fs";
import path from "path";

interface IThemeIcons {
  light: string;
  dark: string;
}

const iconList = new Map<string, IThemeIcons>();
const iconBasePath = path.resolve(__dirname, "../../resources/icons");

fs.readdirSync(iconBasePath).forEach(filename => {
  const iconName = filename.replace(/-(dark|light)\.svg/, "");
  if (iconList.has(iconName)) return;

  iconList.set(iconName, {
    light: path.join(iconBasePath, `${iconName}-light.svg`),
    dark: path.join(iconBasePath, `${iconName}-dark.svg`),
  });
});

if (iconList.size === 0) throw new Error("Failed to create icon list.");

export function getExtensionIcon(iconName: string) {
  const icon = iconList.get(iconName);
  if (!icon) throw new Error("Icon not in icon list.");

  return icon;
}
