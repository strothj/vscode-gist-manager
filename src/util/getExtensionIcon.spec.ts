import fs from "fs";
import { getExtensionIcon } from "./getExtensionIcon";

it("returns extension icon structure for specified icon name", () => {
  const existingIcon = getExtensionIcon("login");

  expect(fs.existsSync(existingIcon.light)).toEqual(true);
  expect(fs.existsSync(existingIcon.dark)).toEqual(true);
});

it("throws error for non-existent icon", () => {
  expect(() => {
    getExtensionIcon("non-existent-icon");
  }).toThrowErrorMatchingSnapshot();
});
