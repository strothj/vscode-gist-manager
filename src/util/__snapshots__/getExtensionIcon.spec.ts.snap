// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws error for non-existent icon 1`] = `"Icon not in icon list."`;
